packages/api/src/base/Decorate.ts

```diff
@@ -454,7 +463,10 @@ export abstract class Decorate<ApiType extends ApiTypes> extends Events {
       ...Array(Math.ceil(keys.length / PAGE_SIZE_VALS))
         .fill(0)
         .map((_, index): Observable<Codec[]> =>
-          this._rpcCore.state.subscribeStorage(keys.slice(index * PAGE_SIZE_VALS, (index * PAGE_SIZE_VALS) + PAGE_SIZE_VALS))
+          ((this.hasSubscriptions || !this._rpcCore.state.queryStorageAt)
```

Comment:
```suggestion
          (this.hasSubscriptions
```

Just one small change to align with `_decorateMulti` above

Reply:
updated